torch/jit/_script.py

```diff
@@ -783,11 +784,36 @@ def call_prepare_scriptable_func_impl(obj, memo):
 
     return obj
 
+
 def call_prepare_scriptable_func(obj):
     memo: Dict[int, torch.nn.Module] = {}
     return call_prepare_scriptable_func_impl(obj, memo)
 
-def script(obj, optimize=None, _frames_up=0, _rcb=None):
+
+def create_script_dict(obj, type_hint=None):
+    """
+    Create a ``torch._C.ScriptDict`` instance with the data from ``obj``. If ``type_hint``
+    is provided, it is used to set the type of the ``ScriptDict``. If not, the type is
+    inferred based on the data inside ``obj``.
+
+    Args:
+        obj (dict): The Python dictionary that is used to initialize the ``ScriptDict``
+                    returned by this function.
+
+    Returns:
+        An instance of ``torch._C.ScriptDict`` that has the same data as ``obj``
+        and can be passed between Python and TorchScript with reference semantics and
+        zero copy overhead.
+    """
+    # TODO: Does it make sense to pass in loc=None here?
```

Comment:
I think so. The only meaningful location to give is "current line", which is naturally available when error is thrown I think.